### PR TITLE
Fix API Status Check for Empty API Key

### PR DESCRIPTION
**Pull request by jac18281828**
Python-SDK subscribes and listens forever even if the apikey is blank or invalid. Someone using the SDK for the first time will likely hit that case and find it difficult to understand why they see no data. We can check and fail on error instead

### CODE REVIEW

File: blocknative/utils.py [hunk 1]
```diff
@@ -34,31 +34,25 @@ def status_error_to_exception(message: dict) -> None:
         SDKError: If there is a non-specific error that is indicated by the server.
     """
 
-    if message["status"] == "ok" or "reason" in message:
+    if message['status'] == "ok" or not "reason" in message:
```
Comment (jac18281828) [inline blocknative/utils.py:37]: @taylorjdawson Please double check the "no" error logic is as it should be here - "not "reason"

Comment (taylorjdawson) [reply blocknative/utils.py]: Reason will be there for non error messages as well so maybe it should be:
```python
if message["status"] == "ok" or not message["reason"]:
```

**Comment by taylorjdawson** [comment]
I am thinking instead of the while loops we send the `checkDappId message and block until we get a response from the server (maybe add a timeout of sorts to prevent hangs) and then once we get a response if it is `ok`  then sends off the rest of the messages in the queue if it is an error then the error gets raised and the program exits

For reference here is an example of the responses:

Error:
```json
{
  "version": 0,
  "serverVersion": "0.122.2",
  "timeStamp": "2021-11-03T06:07:32.662Z",
  "connectionId": "C2-382633bb-9194-47be-83c3-81a7f962a77f",
  "status": "error",
  "raw": "{\"timeStamp\": \"2021-11-03T08:07:19.807813\", \"dappId\": \"12345\", \"version\": \"1\", \"blockchain\": {\"system\": \"ethereum\", \"network\": \"bsc-main\"}, \"categoryCode\": \"initialize\", \"eventCode\": \"checkDappId\"}",
  "event": {
    "timeStamp": "2021-11-03T08:07:19.807813",
    "dappId": "12345",
    "version": "1",
    "blockchain": {
      "system": "ethereum",
      "network": "bsc-main"
    },
    "categoryCode": "initialize",
    "eventCode": "checkDappId"
  },
  "reason": "12345 is not a valid API key"
}
```

Success
```json
 {
  "version": 0,
  "serverVersion": "0.122.2",
  "timeStamp": "2021-11-03T06:10:43.103Z",
  "connectionId": "C9-14df3a03-1bc8-4421-b586-b5eabd8e21af",
  "status": "ok",
  "raw": "{\"timeStamp\": \"2021-11-03T08:10:29.505917\", \"dappId\": \"<redacted>\", \"version\": \"1\", \"blockchain\": {\"system\": \"ethereum\", \"network\": \"bsc-main\"}, \"categoryCode\": \"initialize\", \"eventCode\": \"checkDappId\"}",
  "event": {
    "timeStamp": "2021-11-03T08:10:29.505Z",
    "dappId": "<redacted>",
    "version": "1",
    "blockchain": {
      "system": "ethereum",
      "network": "bsc-main"
    },
    "categoryCode": "initialize",
    "eventCode": "checkDappId"
  },
  "reason": ""
}
```

File: blocknative/utils.py (hunk 1 above)
Comment (jac18281828) [reply blocknative/utils.py]: Updated.

**Comment by jac18281828** [comment]
Yes, I'd consider restructuring the whole loop.    The loop should be while 'is_subscribed' not so that once it is done processing and fully unsubscribed it should exit.    That would be a new feature and not a bug fix so I'd leave that out of this.  We should also add in some tests as mostly everything is untested.    

Another thought I have is about removing the extra queue.    Trio is already queuing its messages.   I think we should handle every event as we receive it and allow the application layer to determine if it wants to add any queuing.     Happy to discuss further at a later time.

For now I'd call this 'ready'


> I am thinking instead of the while loops we send the checkDappId message and block until we get a response from the server (maybe add a timeout of sorts to prevent hangs) and then once we get a response if it is ok then sends off the rest of the messages in the queue if it is an error then the error gets raised and the program exits

File: blocknative/utils.py [hunk 2]
```diff
@@ -34,31 +34,25 @@ def status_error_to_exception(message: dict) -> None:
         SDKError: If there is a non-specific error that is indicated by the server.
     """
 
-    if message["status"] == "ok" or "reason" in message:
+    if message['status'] == "ok" or 'reason' not in message:
```
Comment (taylorjdawson) [inline blocknative/utils.py:37]: Let's do double quotes for consistency
```suggestion
    if message["status"] == "ok" or "reason" not in message:
```

File: blocknative/utils.py [hunk 3]
```diff
@@ -34,31 +34,25 @@ def status_error_to_exception(message: dict) -> None:
...
         return None  # This message does not contain an error
 
-    reason = message["reason"]
-
+    reason = message['reason'].rstrip().lstrip()
```
Comment (taylorjdawson) [inline blocknative/utils.py:40]: Same here
```suggestion
    reason = message["reason"].rstrip().lstrip()
```

**Review by taylorjdawson — CHANGES_REQUESTED** [review]
I just realised that now there's a mix of double & single quotes in the repo, would be probably best to decide on one or the other for consistency but either way let's make sure there all the same 👍🏾

File: blocknative/utils.py (hunk 2 above)
Comment (jac18281828) [reply blocknative/utils.py]: We should standardize on using single quotes as that is more 'pythonic'.   I will revert here but we should follow up in a later pr.

**Review by taylorjdawson — APPROVED** [review]

**Pull request closed without merging**